### PR TITLE
chore(CIS): password expiration, cron file mode enforcement

### LINKED ISSUE #1014: CIS 5.1.2 Ensure permissions on /etc/crontab are configured
```
# 5.1.2 Ensure permissions on /etc/crontab are configured (Scored)

file 'set ownership and permissions on /etc/crontab' do
  path '/etc/crontab'
  mode '0600'
end

# 5.1.3 Ensure permissions on /etc/cron.hourly are configured (Scored)

directory 'Set /etc/cron.hourly' do
  path '/etc/cron.hourly'
  mode '0600'
end

# 5.1.4 Ensure permissions on /etc/cron.daily are configured (Scored)

directory 'Set /etc/cron.daily' do
  path '/etc/cron.daily'
  mode '0600'
end

# 5.1.5 Ensure permissions on /etc/cron.weekly are configured (Scored)

directory 'Set /etc/cron.weekly' do
  path '/etc/cron.weekly'
  mode '0600'
end

# 5.1.6 Ensure permissions on /etc/cron.monthly are configured (Scored)

directory 'Set /etc/cron.monthly' do
  path '/etc/cron.monthly'
  mode '0600'
end

# 5.1.7 Ensure permissions on /etc/cron.d are configured (Scored)

directory 'Set /etc/cron.d' do
  path '/etc/cron.d'
  mode '0600'
end

# 5.1.8 Ensure at/cron is restricted to authorized users (Scored)

file 'Set /etc/at.allow' do
  path '/etc/at.allow'
  mode '0600'
  content "root\n"
end

file 'Set /etc/cron.allow' do
  path '/etc/cron.allow'
  mode '0600'
  content "root\n"
end
```

### LINKED ISSUE #1019: CIS 5.4.1.1 Ensure password expiration is 90 days or less
```
# 5.4.1.1 Ensure password expiration is 90 days or less (Scored)
# 5.4.1.2 Ensure minimum days between password changes is 7 or more (Scored)

cookbook_file 'Set /etc/login.defs' do
  path '/etc/login.defs'
  source 'etc/login.defs'
end
```

### LINKED ISSUE #1020: CIS 5.4.1.4 Ensure inactive password lock is 30 days or less
```
# 5.4.1.4 Ensure inactive password lock is 30 days or less (Scored)

cookbook_file 'Set /etc/default/useradd' do
  path '/etc/default/useradd'
  source 'etc/default/useradd'
end
```

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

Enable opinionated password expiration configuration:

- password max days expiration = 90 days
- password minimum days requirement = 7 days
- inactive passwords locked at 30 days

Enforce `0600` file mode configuration for cron-related files.

**Issue Fixed**:

Fixes #1019
Fixes #1014 
Fixes #1020

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: parts/k8s/cloud-init/artifacts/cis.sh [hunk 1]
```diff
@@ -91,9 +91,28 @@ assignFilePermissions() {
     chmod 644 /etc/default/grub || exit $ERR_CIS_ASSIGN_FILE_PERMISSION
     chmod 644 /etc/pam.d/su || exit $ERR_CIS_ASSIGN_FILE_PERMISSION
     chmod 755 /etc/profile.d/CIS.sh || exit $ERR_CIS_ASSIGN_FILE_PERMISSION
+    for filepath in /etc/crontab /etc/cron.hourly /etc/cron.daily /etc/cron.weekly /etc/cron.monthly /etc/cron.d; do
```
Comment (mboersma) [inline parts/k8s/cloud-init/artifacts/cis.sh]: There is no `/etc/at.allow` or `/etc/cron.allow` I assume?

Comment (jackfrancis) [reply parts/k8s/cloud-init/artifacts/cis.sh]: I think we should defer locking down cron to certain users until we figure out how to do that sustainably. I.e., we don't want to restrict to root, as arguably we *don't* want to encourage folks to use root. Basically, we need a definition of "authorized users".

**Comment by jackfrancis** [comment]
validated on ubuntu 16.04

**Comment by jackfrancis** [comment]
validated on ubuntu 18.04

**Pull request merged**